### PR TITLE
add dev/perforce-testing-helpers to aid with manual Perforce permissions testing

### PR DESCRIPTION
**Pull request by ggilmore**
This PR adds:

- an [example depot](https://github.com/sourcegraph/sourcegraph/tree/example-protects/dev/perforce-testing-helpers/base) that's suitable for manual Perforce permissions testing
- a [script that handles uploading the depot](https://github.com/sourcegraph/sourcegraph/blob/example-protects/dev/perforce-testing-helpers/p4-upload.sh) at the start of a new round of testing
	- See a video of this script running:
		- 	[asciicast](https://asciinema.org/a/UH3gGpXpkSclPBBKWf3lHPA1Z)
- a [script that creates a test user, associated groups, prompts the user to select the groups they'd the test user to be a part of, and applies a protections table](https://github.com/sourcegraph/sourcegraph/blob/example-protects/dev/perforce-testing-helpers/p4-setup-protections.sh) between each test case
	- See a video of this script running:
	  	- [asciicast](https://asciinema.org/a/l3AZaNbEf5nshzFv4JVp5RmgM)

See the [README](https://github.com/sourcegraph/sourcegraph/blob/example-protects/dev/perforce-testing-helpers/README.md) for more information.

## Test plan

These scripts are external helper tools which aren't used by any of our services. However, I have verified that they work via some manual testing and liberal use of `set -x`.

### CODE REVIEW

File: dev/perforce-testing-helpers/templates/p4_protects.tmpl [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	write group Backend * //test-perms/Backend/...	## DO NOT DELETE This is used for backend integration tests
+	=read user alice * -//test-perms/Security/...	## DO NOT DELETE This is used for backend integration tests
+	write group Backend * //test-perms/Security/...
+	list group Everyone * -//...
+	=read group DocSiteDenyPath * -//dax-docsite/cmd/docsite/build.go
+	write group DocSiteWritePath * //dax-docsite/cmd/docsite/build.go
+	list group TestGroup * -//mike-test/...	## edge case testing
+	write group TestGroup * //mike-test/branches/cygnus/boehm-gc/...	## edge case testing
+# AWK-START ## (place protection rules for integration tests below me)
+	=read group all * -//${DEPOT_NAME}/...
```
Comment (ggilmore) [inline dev/perforce-testing-helpers/templates/p4_protects.tmpl:98]: These protection lines between the comments are all replaceable - I just left them here as examples.

**Comment by ggilmore** [comment]
@sourcegraph/repo-management @mollylogue @ryanslade PTAL

**Review by mollylogue — APPROVED** [review]
This looks great!! Thanks for adding the README and for the detailed PR description. It is much appreciated 🙏

**Review by ryanslade — COMMENTED** [review]
Awesome, thanks for doing this!

**Pull request merged**